### PR TITLE
Get rid of "GetLeader" command 

### PR DESCRIPTION
**Pull request by kschrab**
## Type of this PR 

- [x] Bug fix
- [ ] Enhancement

## Description

Removed `GetLeader` command from Traci/Libsumo Bridge Implementation, as we can fetch the leading vehicle using subscriptions. Still, the method Vehicle.getLeader in the Libsumo implementation is not yet usable. 
Furthermore, if no leading vehicle is found, a non-null value is returned (`LeadingVehicle.NO_LEADER`) for better handling.

In earlier versions of SUMO (< 1.0.0) there were no subscriptions available for leading vehicle. To support backward compatibility we kept this command. This is not required anymore.

## Issue(s) related to this PR

 * No issue related to the PR
 
 ## Affected parts of the online documentation

* No

## Definition of Done

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] All commits are signed-off (`-s`) with your mail address of your Eclipse Account.
- [x] `origin/main` has been merged into your Fork.
- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [x] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been observed (see CONTRIBUTING.md).
- [x] All tests pass.

### CODE REVIEW

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java [hunk 1]
```diff
@@ -649,31 +644,25 @@ private VehicleSensorData rearDistance(double rearDistance) {
                 continue;
             }
             if (queryingAllVehiclesRequired || vehiclesWithFrontSensor.containsKey(veh.id)) {
-                try {
-                    LeadingVehicle leader = ((VehicleSubscriptionResult) veh).leadingVehicle != null
-                            ? ((VehicleSubscriptionResult) veh).leadingVehicle
-                            : vehicleGetLeader.execute(bridge, veh.id, Double.MAX_VALUE);
```
Comment (kschrab) [inline fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:655]: This was the initial reason why I removed the command. Even when subscriptions for leading vehicle was active, the "GetLeader" command was always called if no leader was returned in the subscriptions, resulting in many unnecessary TraCI calls.

**Comment by kschrab** [comment]
@rprotzmann this PR is not really required for merging issue 185, but would improve SUMO coupling a bit by removing the GetLeader request and better error handling if SUMO could not be started.

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java [hunk 2]
```diff
@@ -630,12 +626,11 @@ private VehicleSensorData rearDistance(double rearDistance) {
     /**
      * Calculates the sensor data from rear.
      *
-     * @param bridge Connection to Traci.
      * @param subscriptions   Subscription data.
      * @return Calculated sensor data.
      * @throws InternalFederateException if leading vehicle for a vehicle couldn't be read
      */
-    private Map<String, VehicleSensorData> calculateBackSensorData(Bridge bridge, List<AbstractSubscriptionResult> subscriptions) throws InternalFederateException {
+    private Map<String, VehicleSensorData> calculateBackSensorData(List<AbstractSubscriptionResult> subscriptions) {
```
Comment (rprotzmann) [inline fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:633]: It is not clear why the method is called `calculateBackSensorData()` when also `vehiclesWithFrontSensor` are taken into account. 

It seems that this method "calculates" (or better determines) leading vehicles globally and the later called `calculateSensorData()` prepares the global data for each individual vehicle??
Actually, the whole `SimulationFacade` is already quite long and hard to understand for non-insiders. Maybe further partitions would be helpful.

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:633]: SUMO can only return the leader of a vehicle (i.e. front sensor). this method uses the front sensor data of all vehicles to calculate the back sensor data (a vehicle is the follower of its leader). I guess a better comment would be sufficient.

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:633]: But yes, apparently this also calculates the front distance sensor based on leader information, I'll have a look how this method can be renamed.

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java]: I refactored the code quite a bit. Decreased complexity and increased readability 🥳

**Pull request merged**